### ISSUE TITLE
Update documentation to reflect 'ESM first' approach

### ISSUE THREAD

**Issue by jonkoops** [Leaflet/Leaflet#8352]
Now that we have landed support for ES modules in #8329 the documentation should be updated to reflect this as the preferred approach to use Leaflet. This includes:

- Replacing usages of the global `L` with imports (preferably named)
- Updating installation instructions to prefer loading Leaflet as an import from CDNs
- Possibly recommending [Import Maps](https://caniuse.com/import-maps) (needs more browser adoption).

**Comment by patrickarlt** [comment]
I would be happy to take this on if you wanted some help. I've documented other projects with [the SkyPack CDN](https://www.skypack.dev/) which provides a CDN optimized for delivering ES modules. Once a beta/rc is on NPM it should be pretty strightforward to get the doc updated.

**Comment by jonkoops** [comment]
@patrickarlt feel free to give it a first draft if you like, I think it's good to start putting some ideas out there. Most likely this is something we'd try to land in multiple PRs, starting with the [download](https://leafletjs.com/download.html) and [quickstart](https://leafletjs.com/examples/quick-start/) pages.

We'll also have to make some changes so that this documentation is not immediately published to from `main`. Perhaps we'll have to deploy this to another domain (e.g. `next.leafletjs.com`). @mourner @Falke-Design WDYT?

**Comment by Falke-Design** [comment]
Yeah fine for me. But I don't know if it is possible to deploy from different branches to different domains over Github.

Or maybe `v2.leafletjs.com`?

**Comment by jonkoops** [comment]
I think `v2.leafletjs.com` would be nice yeah, @mourner is this something we can arrange?

**Comment by mourner** [comment]
@jonkoops that'd be great! Need to figure out how to do it...

**Comment by IvanSanchez** [comment]
@mourner @jonkoops It seems that we're gonna need a new repo just for `v2.leafletjs.com` (assuming we want to use github pages).

Having several branches of the same repo as different instances of github pages has been asked, and the answer was "no". See https://github.com/orgs/community/discussions/21582 .

It would be posible to do something more convoluted, such as a gitlab mirror of the whole repo, and using gitlab pages for a branch from the mirror. Unfortunately I don't know which approach is the cleanest one / the one with the lowest friction.

**Comment by Falke-Design** [comment]
What if we make a subfolder `leafletjs.com/v2/` instead of a subdomain (`v2.leafletjs.com`)?

**Comment by jonkoops** [comment]
Not sure if that would work. But perhaps we should start deploying the site from the `v1` branch instead? That way we can work on the v2 docs directly on `main`, just like with the other features.

**Comment by Falke-Design** [comment]
Yes I think that would make sense.

@mourner ok for you?

**Comment by mourner** [comment]
@Falke-Design yes! Ideally we would also have a staging URL to check how the v2 website from the `main` looks.

**Comment by jonkoops** [comment]
Ok, I have configured GitHub pages to deploy from the `v1` branch, rather than `main`. So any changes we land in `main` should no longer affect the site.

> yes! Ideally we would also have a staging URL to check how the v2 website from the `main` looks.

Agreed, do we have some way of accomplishing this? It seems GitHub pages will not allow us to do so, should we look for an alternative solution?

**Comment by Falke-Design** [comment]
We could generate the page manually and then push it to the server in a own workflow. With that we could deploy to a subdomain too.

I will mark this with https://github.com/Leaflet/Leaflet/labels/help%20wanted

**Comment by Falke-Design** [comment]
@jonkoops @mourner I never worked with aws, but can we make a second bucket for `v2` and using [this workflow](https://github.com/jakejarvis/s3-sync-action) to push our docs to another aws-bucket?

**Comment by jonkoops** [comment]
I guess we could yeah, @mourner is the owner of that bucket so perhaps he will know. If Github pages doesn't work then it would at least be nice to deploy somewhere else in the mean time.
